pkg/authentication/oidc/authenticate.go

```diff
@@ -0,0 +1,126 @@
+package oidc
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/zitadel/oidc/v3/pkg/client"
+	"github.com/zitadel/oidc/v3/pkg/client/rp"
+	httphelper "github.com/zitadel/oidc/v3/pkg/http"
+	"github.com/zitadel/oidc/v3/pkg/oidc"
+
+	"github.com/zitadel/zitadel-go/v3/pkg/authentication"
+)
+
+type Ctx[C oidc.IDClaims, S rp.SubjectGetter] interface {
+	authentication.Ctx
+	New() Ctx[C, S]
+	SetTokens(*oidc.Tokens[C])
+	GetTokens() *oidc.Tokens[C]
+	SetUserInfo(S)
+	GetUserInfo() S
+}
+
+// CodeFlowAuthentication provides an [authentication.Handler] implementation with
+// an OIDC/OAuth2 Authorization Code Flow.
+// Use [WithCodeFlow] for implementation.
+type CodeFlowAuthentication[T Ctx[C, S], C oidc.IDClaims, S rp.SubjectGetter] struct {
```

Comment:
Struct as private so that you don't have to check in the functions if relyingParty is nil? ( Which is not checked )

Reply:
done